README.md

```diff
@@ -122,6 +122,172 @@ your container. This will allow you to login to the root account remotely. Local
+* `/usr/share/container-layer/mysql/usage/` -- this directory includes text files with extention `.txt` that will be printed as usage message for the container in case wrong arguments are given. Shell variables in these files will be expanded using `envsubst` utility.
+
+Generally, to perform some additional action according to your needs, put appropriate files into these directories in your Dockerfile. See other files in those directories for inspiration.
+
+### Example of extending the Docker image by creating layered Dockerfile
+
+This example adds posibility to add a new user during database inicialization. The username and password is specified using `MYSQL_CONTENT_ADMIN_USER` and `MYSQL_CONTENT_ADMIN_PASSWORD` environment variables, provided when starting the container the first time.
+
+Since you might need to validate the input provided by user in environment variables, it is possible to do it by droping a shell script appropriate directory, that is sourced in the beginning of the daemon start.
+
+Finally, we want to extend the usage message that is printed in case wrong arguments are given by user. This is done by droping a text file into appropriate directory
+
+Content of the layered Dockerfile:
+```
+FROM <this-image>
+USER 0
+ADD 60_content_admin_create.sh /usr/share/container-layer/mysql/post-init/
+ADD 60_content_admin_password.sh /usr/share/container-layer/mysql/pre-start/
+ADD 60_verify_content_admin.sh /usr/share/container-layer/mysql/pre-init/
+ADD 60_usage_content_admin.txt /usr/share/container-layer/mysql/usage/
+USER 27
+```
+
+Content of the `60_content_admin_create.sh` file
+```
+mysql $mysql_flags <<EOSQL
+    CREATE USER '${MYSQL_CONTENT_ADMIN_USER}'@'%' IDENTIFIED BY '${MYSQL_CONTENT_ADMIN_PASSWORD}';
+    GRANT INSERT, SELECT, UPDATE, DELETE ON \`${MYSQL_DATABASE}\`.* TO '${MYSQL_USER}'@'%' ;
+    FLUSH PRIVILEGES ;
+EOSQL
+```
+
+Content of the `60_content_admin_password.sh` file
+```
+mysql $mysql_flags <<EOSQL
+  SET PASSWORD FOR '${MYSQL_CONTENT_ADMIN_USER}'@'%' = PASSWORD('${MYSQL_CONTENT_ADMIN_PASSWORD}');
+EOSQL
+```
+
+Content of the `60_verify_content_admin.sh` file:
+```
+if ! [[ -v MYSQL_CONTENT_ADMIN_USER && -v MYSQL_CONTENT_ADMIN_PASSWORD ]]; then
+    usage
+fi
+
+[ ${#MYSQL_CONTENT_ADMIN_USER} -le 16 ] || usage "MySQL username for MYSQL_CONTENT_ADMIN_USER too long (maximum 16 characters)"
+```
+
+Content of the `60_usage_content_admin.txt` file:
+```
+Content admin user:
+  MYSQL_CONTENT_ADMIN_USER (max 16 characters)
+  MYSQL_CONTENT_ADMIN_PASSWORD
+```
+
+To build such a layered Dockerfile, just run `docker build .` in the directory, that includes all the files above. Built Docker image may be then run as:
+
+```
+docker run -ti --rm -e MYSQL_USER=user -e MYSQL_PASSWORD=pass1 -e MYSQL_DATABASE=db -e MYSQL_CONTENT_ADMIN_USER=contentuser -e MYSQL_CONTENT_ADMIN_PASSWORD=pass2 <image>
```

Comment:
But we're supposed to deliver image usable even without those technologies. We're not blocking you from doing it with OpenShift/Kubernetes template/pod definition but we need something different.